DeepLabCut/deeplabcut_main.py

```diff
@@ -0,0 +1,159 @@
+    if not os.path.exists(labeled_datapath):
+        os.makedirs(labeled_datapath)
+
+    #--------------- Create a pandas DataFrame with body part coordinates (x, y) ---------------
+    header    = []
+    rows      = len(extracted_frames)
+    data:list = [[] for _ in range(rows)]
+    for pose in bodypart_names:
+        header.extend([(experimenter, pose, 'x'),(experimenter, pose, 'y')])
+        label = pose + dlc_defs.Parameters.danse.COORDINATES
+        for i in range(len(extracted_frames)):
+            data[i] += deeplabcut_params.params[label][i]
+
+    columns  = pd.MultiIndex.from_tuples(header, names = ['scorer', 'bodyparts', 'coords'])
+    axis_left = []
+    for frameNum in [int(item) for item in extracted_frames]:
+        axis_left.extend([('labeled-data', video_name, f'img{frameNum}.png')])
+    axis1 = pd.MultiIndex.from_tuples(axis_left)
+    df = pd.DataFrame(data, columns=columns)
+    df.index = axis1
+
+    #--------------- Save dataframe as a hdf file ---------------
+    file_path = f'{labeled_datapath}/CollectedData_{experimenter}.h5'
+    df.to_hdf(file_path, key='keypoints', mode='w')
+
+    #--------------- Save extracted images (used for labeling bodyparts) as .png ---------------
+    cap = cv2.VideoCapture(video_path)
+    frame_count = 0
+    while cap.isOpened():
+        ret, frame = cap.read()
+        if not ret:
+            break 
+        if frame_count in [int(item) for item in extracted_frames]:
+            # Save the frame as a PNG file
+            frame_filename = f'{labeled_datapath}/img{frame_count}.png'
+            cv2.imwrite(frame_filename, frame)
+        frame_count += 1 
+    cap.release() 
+    cv2.destroyAllWindows()
+
+def update_config_file(path_config_file, bodypart_names):
+    # Load the YAML file
+    with open(path_config_file, 'r') as file:
+        data = yaml.safe_load(file)
+
+    # Modify the specific text   
+    data['bodyparts'] = bodypart_names
+
+    # Save the modified YAML back to the file
+    with open(path_config_file, 'w') as file:
+        yaml.safe_dump(data, file, default_flow_style=False)
+
+def save_coords_to_doric(filepath, datapath, path_output, deeplabcut_params, group_names):
+    file_           = h5py.File(filepath, 'a')
+    bodypart_names  = deeplabcut_params.params.pop(dlc_defs.Parameters.danse.BODY_PART_NAMES).split(', ')
+    bodypart_colors = deeplabcut_params.params.pop(dlc_defs.Parameters.danse.BODY_PART_COLORS).split(', ')
+    deeplabcut_params.params[dlc_defs.Parameters.danse.VIDEO_DATAPATH] = datapath
+
+    data, group, operation, series, videoName = group_names
+    time_ = np.array(file_[f"{datapath}/{defs.DoricFile.Dataset.TIME}"])
+    h5_files = glob.glob(os.path.join(path_output, '*.h5'))
+    file_path = h5_files[0]
+    data_coords = pd.read_hdf(file_path)
+    operation_path = f'{group}/{dlc_defs.Parameters.danse.COORDINATES}/{series}/{videoName}{dlc_defs.DoricFile.Group.POSE_ESTIMATION}'
```

Comment:
Please don't forget to enumerate correctly the operation group name, in case the PoseEstimation was applied several times but with different parameters, e.g. review `save_caiman_to_doric()` the lines:
```
operationCount = utils.operation_count(vpath, f, cm_defs.DoricFile.Group.ROISIGNALS, params_doric, params_source)
 params_doric[defs.DoricFile.Attribute.Group.OPERATIONS] += operationCount
```